### PR TITLE
fix: Enable manual input of date range for super admin

### PR DESCRIPTION
**Pull request by ManyRios**
## Describe your changes
  Manually type in the dates in the following format: MM/DD/YYYY and MM/DD/YY
## Issue ticket number and link
 https://github.com/stakwork/sphinx-tribes-frontend/issues/188
## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested on Chrome and Firefox
- [ ] I have tested on a mobile device
- [x] I have provided a screenshot or recording of changes in my PR if there were updates to the frontend

https://github.com/stakwork/sphinx-tribes-frontend/assets/34518489/d81f4a6e-2aa3-4b70-af03-7acdef51bd38

### CODE REVIEW

**Comment by ecurrencyhodler** [comment]
Hey manny. Thanks for submitting this PR. I apologize for the last minute change. But can we format it ti MM/DD/YY instead of YYYY?

**Comment by ManyRios** [comment]
Works for both! You can type ex. 05/27/2023 and also 05/27/23

**Comment by ManyRios** [comment]
What I don't understand if why failed prettier and jest If I always run them before commit

**Comment by Vayras** [comment]
@ManyRios For the fix for the checks you can make these changes --
1. `${StartDate30.format('DD MMM')} - ${expectedEndDate.format('DD MMM YYYY')}` add this instead of `${StartDate30.format('DD MMM YYYY')} - ${expectedEndDate.format('DD MMM YYYY')}` in SuperAdminHeader.spec.tsx line 68 
2. run command yarn run prettier  for prettier fix

**Comment by Vayras** [comment]
> What I don't understand is why failed prettier and jest If I always run them before commit

I think the recents commits merged have changed the format of date hence the test is failing

**Comment by ecurrencyhodler** [comment]
Thanks manny! That's good to know.

**Comment by ManyRios** [comment]
> Thanks manny! That's good to know.


https://github.com/stakwork/sphinx-tribes-frontend/assets/34518489/27a601aa-99ad-4454-ad84-8c92eaf06e5d

File: src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx [hunk 1]
```diff
@@ -5,6 +5,28 @@ import React from 'react';
...
+    }}
+  >
+    <div className="leadingImageContainer">
+      <img className="buttonImage" src="/static/Delete.svg" alt="h" />
+    </div>
+    <button disabled={isDisabled} className="euiText ButtonText css-g2xc3e-euiText-m">
+      {buttonText}
+    </button>
+  </div>
+);
```
Comment (kevkevinpal) [inline src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx:28]: why is this in the spec file?

File: src/people/widgetViews/summaries/wantedSummaries/CodingBounty.tsx
Comment (kevkevinpal) [inline src/people/widgetViews/summaries/wantedSummaries/CodingBounty.tsx:1]: are these changes related to the date change?

File: src/pages/superadmin/header/components/Calendar/App.tsx [hunk 2]
```diff
@@ -10,129 +10,132 @@ interface Props {
   setShowCalendar: (show: boolean) => void;
 }
 
+const Section = styled.div`
```
Comment (kevkevinpal) [inline src/pages/superadmin/header/components/Calendar/App.tsx:13]: we can move these styles to its own CalendarStyles.tsx file and import it here so this tsx file is not as long

File: src/pages/superadmin/header/components/Calendar/App.tsx
Comment (kevkevinpal) [inline src/pages/superadmin/header/components/Calendar/App.tsx:1]: I'm confused why this file is named `App.tsx`

File: src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx (hunk 1 above)
Comment (ManyRios) [reply src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx:28]: this should not be part of this PR it is from the previous PR

File: src/people/widgetViews/summaries/wantedSummaries/CodingBounty.tsx
Comment (ManyRios) [reply src/people/widgetViews/summaries/wantedSummaries/CodingBounty.tsx:1]: No

File: src/pages/superadmin/header/components/Calendar/App.tsx (hunk 2 above)
Comment (ManyRios) [reply src/pages/superadmin/header/components/Calendar/App.tsx:13]: I can do this right now

File: src/pages/superadmin/header/components/Calendar/App.tsx
Comment (ManyRios) [reply src/pages/superadmin/header/components/Calendar/App.tsx:1]: I Dont know, How do you want to rename it?

File: src/people/widgetViews/summaries/wantedSummaries/CodingBounty.tsx
Comment (kevkevinpal) [inline src/people/widgetViews/summaries/wantedSummaries/CodingBounty.tsx:1]: this change doesnt seem relavant to this PR

File: src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx (hunk 1 above)
Comment (kevkevinpal) [reply src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx:28]: can you please remove this from this PR then and do not mark as resolved if it is still in the PR

Comment (ManyRios) [reply src/people/widgetViews/summaries/wantedSummaries/__tests__/CodingBounty.spec.tsx]: Sorry about that

File: src/people/widgetViews/summaries/wantedSummaries/CodingBounty.tsx
Comment (ManyRios) [reply src/people/widgetViews/summaries/wantedSummaries/CodingBounty.tsx:1]: Sorry about that

Comment (ManyRios) [reply src/people/widgetViews/summaries/wantedSummaries/CodingBounty.tsx:1]: Solved

**Comment by kevkevinpal** [comment]
@ManyRios are you able to write a test for this change?

**Comment by ManyRios** [comment]
> @ManyRios are you able to write a test for this change?
Done

**Comment by Vayras** [comment]
https://www.loom.com/share/9763abd132ba4edb909da3691dbeffea @ManyRios  a bug/side effect I have discovered while testing this locally , possible fix I can recommend is when the startdate and enddate input are of the same day, take the startdate with time 00:00:00 so there is a full 24hrs difference between stardate and enddate , aside from that all is good to go. please implement this one thing.

**Comment by ManyRios** [comment]
> https://www.loom.com/share/9763abd132ba4edb909da3691dbeffea @ManyRios a bug/side effect I have discovered while testing this locally , possible fix I can recommend is when the startdate and enddate input are of the same day, take the startdate with time 00:00:00 so there is a full 24hrs difference between stardate and enddate , aside from that all is good to go. please implement this one thing.

Hey @Vayras thanks for your review however your suggestion is original implemented here: 

```typescript
const handleClick = () => {
    if (startDate && endDate) {
      let startDt = startDate;
      let endDt = endDate;
      startDt = set(startDate, { hours: 0, minutes: 0, seconds: 0 });
      if (startDt > endDt) {
        const temp = startDt;
        startDt = endDt;
        endDt = temp;
      }
```
I think the bug it's when there is no bounties in the selected range of dates. In this case... the same day

**Comment by Vayras** [comment]
@ManyRios what I mean is on selecting it from the calendar itself, I can see the bounties as shown in the video, while If I manually input the date I get errors

**Comment by ManyRios** [comment]
> @ManyRios what I mean is on selecting it from the calendar itself I can see the bounties as shown in the video, while If I manually input the date I get errors

Oh ok ok... got it

**Comment by ManyRios** [comment]
> @ManyRios what I mean is on selecting it from the calendar itself I can see the bounties as shown in the video, while If I manually input the date I get errors

https://github.com/stakwork/sphinx-tribes-frontend/assets/34518489/d36756ed-4e58-4301-a9db-f6e23c9868d5

I made a little change, however dont seems very accurate. can you take a look?

**Comment by ManyRios** [comment]
> @ManyRios can you upload a video of you manually adding same date into input for example take "02/06/2023" or "02/06/2024" like I have done in the loom video

take a look in prod 

https://github.com/stakwork/sphinx-tribes-frontend/assets/34518489/d89f36b7-c8b8-47d9-b6b8-51882dd59b51

**Comment by Vayras** [comment]
@ManyRios Worked like a charm, great job. Code is working as Intended tested this locally, once the workflows are approved and passed @kevkevinpal will get this merged

**Comment by ecurrencyhodler** [comment]
Tested and paid. GJ manny!
